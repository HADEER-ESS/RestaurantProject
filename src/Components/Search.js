import React, {useState} from 'react';
import {StyleSheet, TextInput, View, TouchableOpacity} from 'react-native';
import MaterialIcons from 'react-native-vector-icons/dist/MaterialIcons';

const Search = ({fun, text}) => {
  return (
    <View style={styles.searchStyle}>
      <MaterialIcons name="search" color="#000000" size={19} />
      <TextInput
        style={{width: '80%', paddingVertical: 7}}
        value={text}
        onChangeText={txt => fun(txt)}
      />
      <TouchableOpacity onPress={() => fun('')}>
        <MaterialIcons name="close" color="#000000" size={19} />
      </TouchableOpacity>
    </View>
  );
};

export default Search;

const styles = StyleSheet.create({
  searchStyle: {
    flexDirection: 'row',
    alignItems: 'center',
    borderColor: '#000000',
    borderWidth: 2,
    borderRadius: 13,
    paddingHorizontal: 11,
    justifyContent: 'space-between',
    marginVertical: 22,
    height: 37,
    marginHorizontal: 17,
  },
});
