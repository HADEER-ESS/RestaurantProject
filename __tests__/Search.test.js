import React from 'react';
import renderer, { create } from 'react-test-renderer';
import { render, fireEvent ,screen } from '@testing-library/react-native';
import Search from "../src/Components/Search";
import '@testing-library/jest-dom'

// const childrenComponent = jest.fun()
// jest.mock('../src/Components/Search' , () => (props) => {
//     childrenComponent(props)
//     return <mock-childrenComponent/>
// })

describe("Test Search Component" , () => {
    
    it("The Component Render Successfully" , () => {
        const tree = renderer.create(<Search/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it(`Make Sure The Component Accept Props And Display It,
        Text Input Will Accept The Value Text
        And The Function From Higher Component` , () => {
            //Mock The OnChangeText Function
        const mockFunction = jest.fn()

        render(<Search text={'He'} fun={mockFunction}/>)

        //Get The TextInput Component Inside The Search Component
        const textInput = screen.getByPlaceholderText('Enter The Restaurant Name')
                            //Element    Data
        fireEvent.changeText(textInput , 'He')
        expect(textInput.props.value).toBe("He")
        expect(mockFunction).toBeCalled()
    })
})
