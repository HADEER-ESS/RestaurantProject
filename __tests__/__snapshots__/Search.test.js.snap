// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Component The Component Render Successfully 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#000000",
      "borderRadius": 13,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 37,
      "justifyContent": "space-between",
      "marginHorizontal": 17,
      "marginVertical": 22,
      "paddingHorizontal": 11,
    }
  }
  testID="SearchView"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#000000",
          "fontSize": 19,
        },
        undefined,
        {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter The Restaurant Name"
    style={
      {
        "paddingVertical": 7,
        "width": "80%",
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#000000",
            "fontSize": 19,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
